### PR TITLE
feat: use installed community drivers on discovery

### PR DESCRIPTION
**Pull request by leoparente**
- Search and Use installed community drivers on discovery 
- If user specifies a driver that is not listed, give a HINT:

image

### CODE REVIEW

**Comment by leoparente** [comment]
[required] unit test coverage

File: diode-napalm-agent/diode_napalm/cli/cli.py [hunk 1]
```diff
@@ -45,6 +45,12 @@ def run_driver(info: Napalm, config: DiscoveryConfig):
             raise Exception(
                 f"Hostname {info.hostname}: Not able to discover device driver"
             )
+    elif info.driver not in supported_drivers:
+        raise Exception(
+            f"Hostname {info.hostname}: specified driver '{info.driver}' was not found in the current supported drivers list: "
```
Comment (leoparente) [inline diode-napalm-agent/diode_napalm/cli/cli.py]:
```suggestion
            f"Hostname {info.hostname}: specified driver '{info.driver}' was not found in the current installed drivers list: "
```

File: diode-napalm-agent/tests/test_cli.py [hunk 2]
```diff
@@ -353,6 +353,42 @@ def test_run_driver_no_driver(
...
+    mock_get_network_driver.return_value = mock_np_driver
+
+    with pytest.raises(Exception) as excinfo:
+        run_driver(info, config)
+
+    mock_discover_device_driver.assert_not_called()
+    mock_get_network_driver.assert_not_called()
+
+    assert str(excinfo.value).startswith(
+        f"Hostname {info.hostname}: specified driver '{info.driver}' was not found in the current supported drivers list:"
```
Comment (leoparente) [inline diode-napalm-agent/tests/test_cli.py]:
```suggestion
        f"Hostname {info.hostname}: specified driver '{info.driver}' was not found in the current installed drivers list:"
```

File: diode-napalm-agent/diode_napalm/discovery.py [hunk 3]
```diff
@@ -4,13 +4,39 @@
...
+import importlib_metadata
 from napalm import get_network_driver
 
 # Set up logging
 logging.basicConfig(level=logging.INFO)
 logger = logging.getLogger(__name__)
 
-supported_drivers = ["ios", "eos", "junos", "nxos"]
+
+def napalm_driver_list():
```
Comment (mfiedorowicz) [inline diode-napalm-agent/diode_napalm/discovery.py]: nit: return type hint
```suggestion
def napalm_driver_list() -> list[str]:
```

**Review by mfiedorowicz — APPROVED** [review]
Thanks Leo, LGTM, one minor suggestion to consider

**Review by rboucher-me — APPROVED** [review]

**Pull request merged**